PlainTasksDates.py

```diff
@@ -548,19 +548,19 @@ def extract_tag(self, point):
             tag under cursor (i.e. point)
         '''
         start = end = point
-        tag_pattern = r'(?<=\s)(\@[^\(\) ,\.]+)([\w\d\.\(\)\-!? :\+]*)'
-        line = self.view.line(point)
-        matches = re.finditer(tag_pattern, self.view.substr(line))
-        for match in matches:
-            m_start = line.a + match.start(1)
-            m_end   = line.a + match.end(2)
-            if m_start <= point <= m_end:
-                start = line.a + match.start(2)
-                end   = m_end
-                break
-        else:
-            match = None
-        tag = match.group(0) if match else ''
+        tag = ''
+        if 'tag.todo' in self.view.scope_name(point):
+            reg = self.view.extract_scope(point)
```

Comment:
The whole thing is a workaround to get it working with the re module.
Phantoms only appear after @due tags, right? That can be worked around with a one time hack if thats so.

Using view.find will work as long as no phantoms throw off the regex. The key feature missing is the regex recursion that makes it all super easy. Probably a much better solution

A more permanent solution is to remove all phantoms at the beginng of the function and put them back at the end which I think is what the other function does? Means the issue should never resurge. Having a single function to reapply the phantoms or just store them and redraw.

---
Having "due_preview_offset": 1 means that date is inside the brackets while also next to your current text. This works great for `@due(+2)` but less good for any with short dates as you end up with both dates in the bracket, ![](http://i.imgur.com/hDBH3qz.png) (this is fixed in another branch to properly encode the date in dd/mm/yy)

Does the setting need to effect every date? Does look strange with two dates in the brackets but that's why it's a setting.